app/models/openscap_result.rb

```diff
@@ -0,0 +1,68 @@
+require 'openscap'
+require 'openscap/ds/arf'
+require 'openscap/xccdf/benchmark'
+
+class OpenscapResult < ApplicationRecord
+  belongs_to :container_image
+  has_one    :binary_blob,           :dependent => :destroy, :autosave => true, :as => :resource
+  has_many   :openscap_rule_results, :dependent => :destroy, :autosave => true
+
+  OpenSCAP.oscap_init
+
+  def html
+    return unless binary_blob && binary_blob.binary
+    with_oscap_file(binary_blob.binary) do |arf_path|
+      arf = OpenSCAP::DS::Arf.new(arf_path)
+      arf.html
```

Comment:
please run `arf.destroy` here.

Ruby garbage collector will of course destroy the arf object. However, there is no way to tell GC that this object has special destructor.